### DISCUSSION TITLE
<script> tag in react markdown

### DISCUSSION THREAD

**Discussion by HamzaFractl** [remarkjs/remark#1044] [Q&A] [answered]
Is there any way I can support this type of code in react markdown, rehype-raw plugin isn't supporting this type of code.

<script> var myVar1 = setInterval(function() {   myTimer1(); }, 1000);  function myTimer1() {   var d1 = new Date();   document.getElementById("clock1").innerHTML = d1.toLocaleTimeString(); } </script>  <H3 id="clock1"></H3>

**Comment by ChristianMurphy** [comment, marked as answer]
No, React prevents this.
React components and JSX cannot directly declare/invoke script tags, and `react-markdown` is a React component.
https://stackoverflow.com/questions/34424845/adding-script-tag-to-react-jsx

**Reply by ChristianMurphy** [reply]
Though I believe this is an [XY Problem](https://xyproblem.info/).
Reframing the question as: "is there a way to run custom content/components on the page?"
Yes, there are several, such as:
* re-mapping an existing tag to your custom element using `components` https://github.com/remarkjs/react-markdown#appendix-b-components 
* Using Directives https://github.com/orgs/remarkjs/discussions/625
* Using MDX https://mdxjs.com/
